presto-native-execution/presto_cpp/main/operators/StorageBasedLocalShuffle.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/StorageBasedLocalShuffle.h"
+#include <filesystem>
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+
+namespace {
+inline std::string
+createShuffleFilename(std::string rootPath, int32_t partition, int fileIndex) {
+  return rootPath + "/" + std::to_string(partition) + "_" +
+      std::to_string(fileIndex) + ".bin";
+}
+}; // namespace
+
+namespace facebook::presto::operators {
+
+std::unique_ptr<velox::WriteFile> StorageBasedLocalShuffle::getNextOutputFile(
+    int32_t partition) {
+  auto fileCount = getPartitionFilesCount(partition);
```

Comment:
getPartitionFilesCount is not very cheap. Any reason we couldn't maintain the total number of files written to a given partition in a member variable?

Reply:
Thank you for clarifying. Would you add a short comment about this to the code to help future readers understand the reasons for this implementation?